src/app/lib/mainsail/transaction.service.ts

```diff
@@ -284,36 +317,28 @@ export class TransactionService extends Services.AbstractTransactionService {
 		return wallet.nonce().toFixed(0);
 	}
 
-	// @TODO: Implement.
-	// eslint-disable-next-line no-unused-private-class-members
-	#signWithLedger(): SignedTransactionData {
-		let signedTransactionBuilder;
-
-		//const senderPublicKey = await this.#ledgerService.getPublicKey(input.signatory.signingKey());
-		//const extendedPublicKey = await this.#ledgerService.getExtendedPublicKey(input.signatory.signingKey());
-		//const senderAddress = (await this.#addressService.fromPublicKey(extendedPublicKey)).address;
-		//
-		//const serialized = await this.#app.resolve(Utils).toBytes(transaction.data);
-		//const signature = await this.#ledgerService.sign(input.signatory.signingKey(), serialized.toString("hex"));
-		//
-		//transaction.data = {
-		//	...transaction.data,
-		//	...signature,
-		//	senderAddress,
-		//	senderPublicKey,
-		//	v: Number.parseInt(signature.v) + 27, // @TODO: remove it on mainsail evm.16
-		//};
-		//
-		//// Reassign public key to match the signer, as `build` changes it.
-		//const signedTransaction = await transaction.build(transaction.data);
-		//signedTransaction.data.senderPublicKey = senderPublicKey;
-		//signedTransaction.data.senderAddress = senderAddress;
-		//
-		//return new SignedTransactionData().configure(signedTransactionBuilder.transaction.data.id, signedTransactionBuilder.transaction.data, signedTransactionBuilder.transaction.data, signedTransactionBuilder.transaction.serialize().toString("hex"))
+	async #sign(input: Services.TransferInput, builder: any): Promise<void> {
+		const { address } = await this.#signerData(input);
+		builder.transaction.data.senderAddress = address;
 
-		return new SignedTransactionData().configure(
-			signedTransactionBuilder.transaction.data,
-			signedTransactionBuilder.transaction.serialize().toString("hex"),
+		if (input.signatory.actsWithLedger()) {
+			return this.#signWithLedger(input, builder.transaction);
+		}
+
+		await builder.sign(input.signatory.signingKey());
+	}
+
+	async #signWithLedger(input: Services.TransferInput, transaction: any): Promise<void> {
+		const signature = await this.#ledgerService.sign(
+			input.signatory.signingKey(),
+			transaction.serialize().toString("hex"),
 		);
+
+		transaction.data = {
+			...transaction.data,
+			...signature,
+			id: transaction.getId(),
+			v: Number.parseInt(signature.v) + 27,
```

Comment:
```suggestion
			v: Number.parseInt(signature.v),
```

this can be removed now as the latest mainsail version no longer requires it